internal/model/rwfolder.go

```diff
@@ -644,7 +644,16 @@ func (p *rwFolder) handleDir(file protocol.FileInfo) {
 			if err != nil || p.ignorePermissions(file) {
 				return err
 			}
-			return os.Chmod(path, mode)
+
+			// Stat the directory so we can check its permissions.
+			info, err := osutil.Lstat(path)
+			if err != nil {
+				return err
```

Comment:
Actually, this whole section should be unnecessary, as the directory is newly created and presumably has the bits set already.